### PR TITLE
Add documentation about release process and the automation to generate changelog

### PR DESCRIPTION
**Pull request by rexwangcc**
### Purpose

- No ticket is linked to this PR. But this is stated as one of the issues in [What the WFL doc](https://broadinstitute.atlassian.net/wiki/spaces/GHConfluence/pages/1671364672/What+the+WFL)

### Changes

- [x] Add a command to cli.py to generate changelog for us.
- [x] Use the new command to generate the current changelog for us.
- [x] Add docs about releases. 
- [x] ~Make the release Action use this new command.~ Instead, the generation of changelog will happen in the "release" PR to provide more control to the devs.
- [x] Clean up the main readme and redirect people to the docsite so the main readme syntax does not confuse people. This is another item in [What the WFL doc](https://broadinstitute.atlassian.net/wiki/spaces/GHConfluence/pages/1671364672/What+the+WFL)

### Review Instructions

- No instructions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Release 0.4.0
```
Comment (ehigham) [inline CHANGELOG.md:1]: This hasn't been released yet - perhaps this can be part of the release action to generate the file?

Comment (rexwangcc) [reply CHANGELOG.md:1]: Oh thanks for catching it! Yeah, if that's the case, let me take this file out. Sounds good, I can let the `cli.py` touch this file before trying to open, and Action can just use the `cli.py`'s `release` command.

File: ops/cli.py [hunk 2]
```diff
@@ -279,7 +280,41 @@ def check_git_tag(config: WflInstanceConfig) -> None:
...
 
 
+def get_git_commits_since_last_tag(config: WflInstanceConfig) -> None:
+    "Read commit messages since last tag, store to config and print."
+    command = 'git log --pretty=format:"%s" $(git describe --tags --abbrev=0 @^)..@'
+    info("=>  Reading commit messages from git log")
+    lines = subprocess.check_output(command, shell=True, encoding="utf-8").strip().split("\n")
+    info("=>  Markdown-rize log messages")
+    current_changelog = "\n".join([f"- {line}" for line in lines])
+    config.current_changelog = current_changelog
```
Comment (jack-r-warren) [inline ops/cli.py:290]: Should/could we do a regex here to replace `#123` with a link to that PR?

**Review by jack-r-warren — APPROVED** [review]

**Comment by jack-r-warren** [comment]
(Review approval based on Ed's comment being addressed, just wanted to communicate that my suggestion was in no way blocking)

File: ops/cli.py (hunk 2 above)
Comment (rexwangcc) [reply ops/cli.py:290]: Sounds like a good idea, I can try to see if it's feasible with simple regex.

Comment (rexwangcc) [reply ops/cli.py]: Pushed new changes, I'm no expert on `regex`, lemme know if that's over comlicated! @jack-r-warren

File: ops/cli.py [hunk 3]
```diff
@@ -279,7 +281,50 @@ def check_git_tag(config: WflInstanceConfig) -> None:
...
+    marked_commit = regex.sub(f"[\g<0>](https://github.com/broadinstitute/wfl/pull/{num_pr[1:]})", commit)
+    marked_commit = f'- {marked_commit}'
+    return marked_commit
+
+def get_git_commits_since_last_tag(config: WflInstanceConfig) -> None:
+    "Read commit messages since last tag, store to config and print."
+    command = 'git log --pretty=format:"%s" $(git describe --tags --abbrev=0 origin/master^)..origin/master'
+    info("=>  Reading commit messages from git log")
+    lines = shell(command).split("\n")
+    info("=>  Markdown-rize log messages")
```
Comment (rexwangcc) [inline ops/cli.py:297]:
```suggestion
    info("=>  Markdown-ify log messages")
```

File: docs/md/dev-release.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+1. Developers create a standalone "release PR",
+which should be exclusively used by releasing WFL.
+2. In the PR, developers need to semanticaly bump
+the version string in the `/version` file at the root
+of the repo.
+3. In the PR, developers need to run `./ops/cli.py release`
+to generate the changelog for this release. (`-d` flag can
+be used to do a dry run without writing to the `CHANGELOG.md`
+file)
+4. Once the release PR is reviewed by merged to `master`
```
Comment (jack-r-warren) [inline docs/md/dev-release.md:15]:
```suggestion
4. Once the release PR is reviewed and merged to `master`
```

**Review by jack-r-warren — APPROVED** [review]
This looks great. Very much appreciate cleaning up the README linking and stuff too, thanks for doing this

**Review by ehigham — APPROVED** [review]

**Pull request merged**